File: Programmers_Quiz/1.주식가격.py
def solution(prices):
    answer = []

    for i in range(len(prices)):
        temp = 0
        for j in range(i+1, len(prices)):
            if prices[i] <= prices[j]:
                temp += 1
            else:
                temp += 1
                break
        answer.append(temp)

    return answer


a = [1, 2, 3, 2, 3]
b = solution(a)
for i in b:
    print(i)


##### brute force #####

# def solution(prices):
#     answer = []

#     for i in range(len(prices)):
#         temp = 0
#         for j in range(i+1, len(prices)):
#             if prices[i] <= prices[j]:
#                 temp += 1
#             else:
#                 temp += 1
#                 break
#         answer.append(temp)

#     return answer


# a = [1, 2, 3, 2, 3]
# b = solution(a)
# for i in b:
#     print(i)

File: Programmers_Quiz/1.주식가격.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> prices)
{
    vector<int> answer;
    for (int i = 0; i < prices.size(); i++)
    {
        int temp = 0;
        for (int j = i + 1; j < prices.size(); j++)
        {
            if (prices[i] <= prices[j])
            {
                temp += 1;
            }
            else
            {
                temp += 1;
                break;
            }
        }
        answer.push_back(temp);
    }
    return answer;
}

int main()
{
    vector<int> v{1, 2, 99, 2, 3};
    vector<int> ans = solution(v);
    for (int i = 0; i < ans.size(); i++)
    {
        printf("%d ", ans[i]);
    }
}